data-structures.md

```diff
@@ -107,24 +109,45 @@ type Actor struct {
 	Head    Cid
     
     // Nonce is a counter of the number of messages this actor has sent
-	Nonce   Integer
+	Nonce   Uint64
     
     // Balance is this actors current balance of filecoin
-	Balance AttoFIL
+	Balance BigInteger
 }
 ```
 
+## State Tree
 
+The state trie keeps track of all state in Filecoin. It is effectively a map of addresses to `actors` in the system. It is implemented using a HAMT.
 
+## HAMT
 
-### Serialization
+TODO: link to spec for our CHAMP HAMT
 
-Actors are currently serialized simply by CBOR marshaling them, using lower-camel-cased field names.
 
-## State Tree
 
-The state trie keeps track of all state in Filecoin. It is effectively a map of addresses to `actors` in the system. It is implemented using a HAMT.
+# Filecoin Compact Serialization
 
-## HAMT
+Datastructures in Filecoin are encoded as compactly as is reasonable. At a high level, each object is converted into an ordered array of its fields (ordered by their appearance in the struct declaration), then CBOR marshaled, and prepended with an object type tag.
+
+| object | tag  |
+|---|---|
+| block v1 | 43  |
+| message v1 | 44 |
+| signedMessage v1 | 45 |
+| actor v1 | 46 |
+
+For example, a message would be encoded as:
+
+```cbor
+tag<44>[msg.To, msg.From, msg.Nonce, msg.Value, msg.Method, msg.Params]
+```
+
+Each individual type should be encoded as specified:
 
-TODO: link to spec for our CHAMP HAMT
\ No newline at end of file
+| type | encoding |
+| --- | ---- |
+| Uint64 | CBOR major type 0 |
```

Comment:
one more question: what happened to leb128?

Reply:
If CBOR has its own way of encoding everything, I figure it makes the most sense to use that instead, as it will be more compact (numbers under a certain size take up zero extra space over the cost of the type tag).